pyfar/signals/deterministic.py

```diff
@@ -194,6 +211,102 @@ def linear_sweep_time(n_samples, frequency_range, n_fade_out=90, amplitude=1,
+
+    .. note::
+        The envelope of the sweep time signal should be constant, apart from
+        slight overshoots at the beginning and end. If this is not the case,
+        try to increase `n_samples`, `start_margin`, `stop_margin`, `fade_in`
+        or `fade_out`.
+
+    Parameters
+    ----------
+    n_samples : int
+        The length of the sweep in samples.
+    frequency_range : array like
+        Frequency range of the sweep given by the lower and upper cut-off
+        frequency in Hz. The restriction of the frequency range is realized
+        by applying a Butterworth band-pass with the specified frequencies.
+    start_margin : int, float
+        The time in samples, at which the sweep starts. The start margin is
+        required because the frequency domain sweep synthesis has pre-ringing
+        in the time domain.
+    stop_margin : int, float
+        Time in samples, at which the sweep stops. This is relative to
+        `n_samples`, e.g., a stop margin of 100 samples means that the sweep
+        ends at sample ``n_samples-100``. This is required, because the
+        frequency domain sweep synthesis has post-ringing in the time domain.
+    fade_in : int
+        Duration of a squared sine fade-in in samples. The fade starts at the
+        first sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``0`` does not apply
+        a fade-in.
+    fade_out : int
+        Duration of a squared cosine fade-out in samples. The fade ends at the
+        last sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``0`` does not apply
+        a fade-out.
+    butterworth_order : int, None
+        The order of the Butterworth filters that are applied to limit the
+        frequency range by a high-pass if ``frequency_range[0] > 0`` and/or by
+        a low-pass if ``frequency_range[1] < sampling_rate / 2``. The default
+        is ``8``.
+    sampling_rate : int
+        The sampling rate in Hz. The default is ``44100``.
+
+    Returns
+    -------
+    sweep : Signal
+        The sweep signal. The Signal is in the time domain, has a maximum
+        absolute amplitude of 1 and the ``rms`` FFT normalization
+        (see :py:func:`~pyfar.dsp.fft.normalization`).
+    group_delay_sweep : FrequencyData
+        The group delay of the sweep in seconds as a single sided spectrum.
+
+    References
+    ----------
+    .. [#] S. Müller, P. Massarani. 'Transfer Function Measurement with Sweeps.
+           Directors Cut Including Previously Unreleased Material and some
+           Corrections. J. Audio Eng. Soc. 2001, 49 (6), 443-471.
+
+    Examples
+    --------
+    Linear sweep between 50 and 22050 Hz
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> sweep, _ = pf.signals.linear_sweep_freq(
+        ...     2**16, [50, 22050], 1000, 1000)
```

Comment:
```suggestion
        >>> sweep, _ = pf.signals.linear_sweep_freq(
        ...     2**16, [50, 10000], 1000, 1000)
```
maybe the user can see more if its just to 10k

Reply:
I tried as suggested and also tried to decrease the duration, but it doesn't really help. If the time is too short, the temporal envelope gets funny. I would suggest to leave it as it is...